dsp/primitives/demonstrate.py

```diff
@@ -74,85 +79,78 @@ def do_augment(train, k=None, return_all=False):
     return do_augment
 
 
-def sample(train, k: int):
-    branch_idx = (
-        hasattr(dsp.settings, 'branch_idx')
-        and isinstance(dsp.settings.branch_idx, int)
-        and dsp.settings.branch_idx
-    )
-    branch_idx = branch_idx or 0
-    # print(f"branch_idx = {branch_idx}")
-
-    rng = random.Random(branch_idx)
-    shuffled_train = list(train)
+def sample(train: list[Example], k: int):
+    """Sample k examples from train."""
+    rng = random.Random(dsp.settings.branch_idx)
```

Comment:
reviewer note: branch_idx default value is set to 0 in `dsp.settings.brach_idx` during init, therefore the previous checks aren't necessary.